core/domain/base_model_validators.py

```diff
@@ -97,9 +98,17 @@ def __init__(self, field_name, model_class, model_ids):
                 'When fetching instances of UserSettingsModel, please use ' +
                 'UserSettingsModelFetcherDetails instead of ' +
                 'ExternalModelFetcherDetails')
+        filtered_model_ids = []
+        for model_id in model_ids:
+            if not model_id:
+                raise Exception(
+                    'A model id in the field \'%s\' '
+                    'is empty' % field_name)
+            else:
```

Comment:
Do we need else here? (Assuming the top `if` as a blocker.)

Reply:
I have removed the if.